README.md

```diff
@@ -1,15 +1,203 @@
-# An alternative http library and web framework inspired by hyper
+<h1 align="center">Xitca-Web</h1>
+<p align="center">
+  <b align="center"><a href="README.md">Readme</a></b> |
+  <b><a href="https://github.com/HFQR/xitca-web">GitHub</a></b> |
+  <b><a href="https://docs.rs/xitca-web/latest/xitca_web/">Documentation</a></b>
+  <br /><br />
+  <a href="#">
+    <img
+      alt="GitHub code size in bytes"
+      src="https://img.shields.io/github/languages/code-size/HFQR/xitca-web?style=flat-square"
+    />
+  </a>
+  <a href=""
+    ><img
+      alt="Maintenance"
+      src="https://img.shields.io/maintenance/yes/2024?style=flat-square"
+    />
+  </a>
+  <br />
+  <br />
+  <i>
+    A HTTP library and web framework written in 100% safe Rust with zero-copy
+    serialization/deserilization support
+  </i>
+</p>
 
-## Minimum Supported Rust Version
-- 1.79 [^1]
+<br />
+<img src="./images/server.jpg" alt="An image of a server" align="center" />
+<br />
 
-## Motivation
-- Less synchronization and thread per core design is used.
-- 100% safe Rust.[^2]
-- Low memory footprint. Avoid (pre)allocation when possible.
-- Lightweight dependency tree. Avoid adding unnecessary import when possible. Prefer no proc macro code generation when possible(proc macro feature are still offered as opt-in instead of opt-out).
-- Compact and simple code base. Reduce the barrier of understanding of source code for easier contributing.
-- Good interop with other crates. `tokio`(for async runtime) and `http`(for http types) are used directly as dependency.
+<details>
+  <summary><b>Table of Contents</b></summary>
+  <p>
 
-[^1]: `http-file` excluded due to design issue.
-[^2]: only guaranteed inside project's own code. unsafe can and is used by dependencies
+  - [🚀 Features](#features-)
```

Comment:
> I believe we should not include emojis in the readme. They are fun but can be seen as less serious. Boring and plain is preferred.

No I believe emojis can grab the user visually which is great as well. Also, for this I would suggest you can go through many popular projects on the trending they do this quite often. So I don't think it hurts in anyway. What do you think? :slightly_smiling_face: .

Reply:
I get what you mean and I appreciate the thought you put into it. 

That said `xitca-web` is not aiming for popularity and we have zero interest in "hyper" or "trend". Being plain and boring is an explicit choice from the start for the tone of the project.

Reply:
> I get what you mean and I appreciate the thought you put into it.
> 
> That said `xitca-web` is not aiming for popularity and we have zero interest in "hyper" or "trend". Being plain and boring is an explicit choice from the start for the tone of the project.

Also, I will point out that having emojis in this secton also doesn't matter because this will fold up anyway so if will remain hidden. Additionally I am not aiming at trending though but I feel having this can grab the reader's attention and also attracts contribution like it gives the impression like `wow such a cool project should try it out or should contribute to it`. But anyways since you insist I will remove it. :slightly_smiling_face: